AutomatedTesting/Gem/PythonTests/AWS/common/constants.py

```diff
@@ -8,11 +8,15 @@
 import os
 import platform
 
+# To avoid resource name length issues, potentially verbose pipeline names are capped at 25 chars.
+# TODO: consolidate project name formulation for tests and deploy scripts to same place.
+MAX_PIPELINE_NAME_LENGTH = 25
+
 # ARN of the IAM role to assume for retrieving temporary AWS credentials
 ASSUME_ROLE_ARN = os.environ.get('ASSUME_ROLE_ARN', 'arn:aws:iam::645075835648:role/o3de-automation-tests')
 # Name of the AWS project deployed by the CDK applications
 AWS_PROJECT_NAME = os.environ.get('O3DE_AWS_PROJECT_NAME').upper() if os.environ.get('O3DE_AWS_PROJECT_NAME') else \
-    (os.environ.get('BRANCH_NAME', '') + '-' + os.environ.get('PIPELINE_NAME', '') + '-' + platform.system()).upper()
+    (os.environ.get('BRANCH_NAME', '') + '-' + os.environ.get('PIPELINE_NAME', '')[:MAX_PIPELINE_NAME_LENGTH] + '-' + platform.system()).upper()

```

Comment:
Would there also be an issue if the branch name also reaches the char limit?

Reply:
Yes, the ceiling is a 40 char total length which would include everything (branch, project, platform). Capping what is currently the most verbose parameter (Pipeline) is just the most impactful single preventative measure we could add.

As @lmbr-pip says above, long-term there's more work to be done here, including consolidating how we're formulating the project names to one place. At that point we should probably enforce a formula that caps each argument so some portion of the allowed length.